### ISSUE TITLE
Leveraging Vision-Centric Multi-Modal Expertise for 3D Object Detection

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#60]
### Paper Link
https://proceedings.neurips.cc/paper_files/paper/2023/hash/79206ac5b7e88eeeed74997f3b6f4c7f-Abstract-Conference.html

### Code Link

https://github.com/OpenDriveLab/Birds-eye-view-Perception

## Motivation

<img width="400" alt="image" src="https://github.com/user-attachments/assets/b818b55a-4b1c-4b15-a4a8-5f2bb1e6cc5b">

### Vision-Centric Expert (VCD-E): Multi-Modal (Depth only)

1. Previous fusion methods that adopt **two modality-specific backbones**, we only leverage a single branch to generate semantic features based on image input, and point clouds **only provide depth information**.
2. VCD-E (vision-centric expert) eliminates the need for intricate training strategies or specialized fusion module designs while reaching a comparable performance to current fusion methods.
3. More importantly, the vision-centric multi-modal model has the exact **same architecture** as the camera-only apprentice model, and the generated BEV spatial representation is solely from image features.

### Vision-Centric Apprentice (VCD-A): Camera-only, _Distilled_ from VCD-E

A trajectory-based distillation module (for dynamic objects) and occupancy reconstruction (for depth prediction).

### Overall Architecture
- The expert and apprentice models adopt the **consistent model architecture**. The **only difference** is that the expert additionally leverages the accurate **depth map generated from the point cloud**, while the apprentice model predicts the depth map from the image.
- We construct a distillation framework between the expert network and the apprentice network.
- After obtaining the pretrained vision-centric expert and corresponding apprentice network, we **freeze the expert** network and leverage its **intermediate features as auxiliary supervision for the apprentice network**.

<img width="1459" alt="image" src="https://github.com/user-attachments/assets/12341c2a-92d5-4f89-ae6c-d383cb91174b">

### The Generation of Expert Model

<img width="1203" alt="image" src="https://github.com/user-attachments/assets/5dee3f57-4f82-4a2a-aab6-5fa5af0497f0">

> No much details about VCD-E.  Directly use: [BEVDepth: Acquisition of Reliable Depth for Multi-View 3D Object Detection
](https://arxiv.org/abs/2206.10092)
> Code for concat historical BEV features: https://github.com/OpenDriveLab/Birds-eye-view-Perception/blob/a0b2d7296d331a1a113ba3d6747a59e503115523/nuScenes_playground/VCD/mmdet3d/models/detectors/vcd.py#L111 

<img width="700" alt="image" src="https://github.com/user-attachments/assets/909beec1-76e5-4bdb-b069-d796461f1fa7">

> Difference? 1. temporal modeling 2. BEVDepth is camera-only for inference, VCD-E uses projected depth + predicted depth

<img width="700" alt="image" src="https://github.com/user-attachments/assets/710b79ef-8ff2-4c51-b38d-e56f047ef63f">

### Trajectory-based Distillation

1. Determine the actual ego motion matrix between the current frame and each historical frame
2. Apply the ego-motion transformation matrix to the **ground truth object positions (x, y, z)** to obtain the transformed positions in the current frame’s coordinate system
3. Construct the motion trajectories from all historical frames
4. Sample BEV features from expert and apprentice
5. Trajectory-based distillation loss

<img width="500" alt="image" src="https://github.com/user-attachments/assets/3d67785a-5147-434b-8a0d-3e18cb201f2c">

### Occupancy Reconstruction (for Distillation)

1. Utilize depth estimation modules to predict the depth for each image pixel
2. The depth map is back-projected into a 3D point cloud, each image pixel (u, v) is transformed into a 3D coordinate (x, y, z).
3. Then the scores from different pixels in the multi-camera, within the same occupancy region, are accumulated to make decisions about the presence of an object in that region. In this way, we can generate the occupancy O_e and O_a for expert and apprentice model, respectively.

**Why occupancy reconstruction?** The occupancy reconstruction module improves the model’s capability to discern the 3D geometric properties of objects. The grid-based supervisory signals effectively direct the model to enhance its prediction accuracy for object depths. Different voxels within the occupancy structure aggregate the fused depth distribution from various perspective views, thereby enhancing robustness against depth errors.

<img width="700" alt="image" src="https://github.com/user-attachments/assets/8fc0cf50-9d53-4173-ba05-027dec5f02e3">

### Training Loss for Distillation

<img width="700" alt="image" src="https://github.com/user-attachments/assets/048f205e-6e8a-458b-8c22-3c1dc6a9eeb6">

## Experiments 
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/ed0915aa-eb95-49f3-962f-972afac4b772">

<img width="1147" alt="image" src="https://github.com/user-attachments/assets/379c89a2-065e-4df0-904d-97e993b2e7b5">

- **VCD-E does not require any complex fusion strategies or three-stage training schedule**, yet it achieves 67.7% mAP and a 71.1% NDS
- Despite employing only a **single image backbone**, the model’s performance is comparable to state-of-the-art methods such as BEVFusion [38], and it even surpasses UVTR [33] by 2.3% mAP and 0.9% NDS, which utilizes two backbones, by a considerable margin. 
- Owing to its homogeneous nature and compatibility with camera-only models, the VCD-E can serve as a **powerful expert** for knowledge transfer, facilitating performance improvements for small to large vision detectors.

<img width="1159" alt="image" src="https://github.com/user-attachments/assets/20c4d817-067a-4936-b1cb-a88fd625d1d3">

- Table 4: For the apprentice models, we incorporate a sequence of **eight frames** into the temporal modeling process. In contrast, the expert models integrate **four future frames** into the temporal modeling, as demonstrated in our primary results.
- Table 5: Homogeneous expert models significantly influence the success of knowledge transfer.
- Table 6:  For fair comparison, we conduct all experiments (distillation strategies) based on our vision-centric expert VCD-E.

<img width="1115" alt="image" src="https://github.com/user-attachments/assets/58b51e70-1c24-489c-b54f-628ac8e1de86">

**Table 8 - Fusion Strategy of Expert**

1.  Predicted depth that directly uses predicted depth based on image features. 
2. LiDAR depth that utilizes the projected depth from LiDAR points. 
3. Fusion depth that employs the projected depth when corresponding LiDAR points exist for each pixel, otherwise using the predicted scores based on image features,
4. Weighted depth. Applying the weighted average depth of the predicted depth and LiDAR depth. 

> What happens if corresponding LiDAR points doe NOT exist for one pixel? How to use strategy 2?

<img width="1189" alt="image" src="https://github.com/user-attachments/assets/04753a8e-074a-49dd-b18d-2b918a11a578">

<img width="964" alt="image" src="https://github.com/user-attachments/assets/fb3af65d-a1f1-47c7-9285-362a3c749986">

<img width="755" alt="image" src="https://github.com/user-attachments/assets/10a41e58-9926-479d-a6d9-2bd307b9d722">
